internal/raft/statemachine.go

```diff
@@ -0,0 +1,108 @@
+package raft
+
+import (
+	"encoding"
+	"fmt"
+	"io"
+
+	"github.com/lni/dragonboat/v4/statemachine"
+)
+
+// Event to update the state machine. These are stored in the Raft log.
+type Event interface {
+	encoding.BinaryMarshaler
+}
+
+// Unmasrshallable is a type that can be unmarshalled from a binary representation.
+type Unmasrshallable[T any] interface {
```

Comment:
There is a typo in masrsable.

Reply:
I could swear this was written by an AI. Maybe it has learned my typo habits.
Will fix!